### PR TITLE
Add WildFly on Standalone CentOS 8 VM

### PR DESCRIPTION
**Pull request by Suraj2093**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Add WildFly on Standalone CentOS 8 VM

### CODE REVIEW

File: wildfly-standalone-centos8/scripts/wildfly-setup.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+echo "tar xvf wildfly-$WILDFLY_RELEASE.Final.tar.gz" | adddate >> wildfly.install.log
+tar xvf wildfly-$WILDFLY_RELEASE.Final.tar.gz | adddate >> wildfly.install.log 2>&1
+
+echo "echo "AllowTcpForwarding no" >> /etc/ssh/sshd_config" | adddate >> wildfly.install.log
+echo "AllowTcpForwarding no" >> /etc/ssh/sshd_config | adddate >> wildfly.install.log 2>&1
+echo "systemctl restart sshd" | adddate >> wildfly.install.log
+systemctl restart sshd | adddate >> wildfly.install.log 2>&1
+
+echo "Sample app deploy..." | adddate >> wildfly.install.log
+echo "git clone https://github.com/Suraj2093/dukes.git" | adddate >> wildfly.install.log
```
Comment (bmoore-msft) [inline wildfly-standalone-centos8/scripts/wildfly-setup.sh:36]: you cannot pull from a personal repo in a QuickStart sample

File: wildfly-standalone-centos8/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        "description": "The size of the Virtual Machine"
+      }
+    }
+  },
+  "variables": {
+    "singlequote": "'",
+    "imagePublisher": "OpenLogic",
+    "imageOffer": "CentOS",
+    "imageSKU": "8.0",
+    "nicName": "[concat(uniquestring(resourceGroup().id), '-nic')]",
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply wildfly-standalone-centos8/azuredeploy.json]: This is changed to "nicName": "centos-vm-nic"

File: wildfly-standalone-centos8/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        ]
+      }
+    },
+    "subnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworkName'), variables('subnetName'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2019-05-01",
+      "name": "pid-2c0c0514-a608-492e-b64e-bc4ace59035b",
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply wildfly-standalone-centos8/azuredeploy.json:111]: Yes this is registered

File: wildfly-standalone-centos8/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+          "networkInterfaces": [
+            {
+              "id": "[resourceId('Microsoft.Network/networkInterfaces',variables('nicName'))]"
+            }
+          ]
+        },
+        "diagnosticsProfile": {
+          "bootDiagnostics": {
+            "enabled": true,
+            "storageUri": "[reference(variables('bootStorageAccountName'), '2017-10-01').primaryEndpoints['blob']]"
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply wildfly-standalone-centos8/azuredeploy.json]: Updated the apiVersion to '2019-06-01'

File: wildfly-standalone-centos8/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+      "properties": {
+        "publisher": "Microsoft.Azure.Extensions",
+        "type": "CustomScript",
+        "typeHandlerVersion": "2.0",
+        "autoUpgradeMinorVersion": true,
+        "settings": {
+          "fileUris": [
+            "[uri(parameters('_artifactsLocation'), concat('scripts/wildfly-setup.sh', parameters('_artifactsLocationSasToken')))]"
+          ],
+          "commandToExecute": "[concat('sh wildfly-setup.sh',' ',parameters('wildflyUserName'),' ',variables('singlequote'),parameters('wildflyPassword'),variables('singlequote'))]"
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply wildfly-standalone-centos8/azuredeploy.json:255]: This is fixed

File: wildfly-standalone-centos8/scripts/wildfly-setup.sh (hunk 1 above)
Comment (Suraj2093) [reply wildfly-standalone-centos8/scripts/wildfly-setup.sh]: Moved this Sample App under the scripts folder and made the corresponding changes in the script

File: wildfly-standalone-centos8/scripts/wildfly-setup.sh [hunk 6]
```diff
@@ -1,67 +1,67 @@
...
+echo "tar xvf wildfly-$WILDFLY_RELEASE.Final.tar.gz" | adddate >> wildfly.install.log
+tar xvf wildfly-$WILDFLY_RELEASE.Final.tar.gz | adddate >> wildfly.install.log 2>&1
+
+echo "echo "AllowTcpForwarding no" >> /etc/ssh/sshd_config" | adddate >> wildfly.install.log
+echo "AllowTcpForwarding no" >> /etc/ssh/sshd_config | adddate >> wildfly.install.log 2>&1
+echo "systemctl restart sshd" | adddate >> wildfly.install.log
+systemctl restart sshd | adddate >> wildfly.install.log 2>&1
+
+echo "Sample app deploy..." | adddate >> wildfly.install.log
+echo "git clone https://github.com/Azure/azure-quickstart-templates.git" | adddate >> wildfly.install.log
```
Comment (bmoore-msft) [inline wildfly-standalone-centos8/scripts/wildfly-setup.sh:36]: We need to handle the WAR file the same way we handle the script itself - by downloading from the staged location.  See: https://github.com/Azure/azure-quickstart-templates/blob/master/100-marketplace-sample/scripts/copyfilefromazure.sh for an example.

Shouldn't need to clone the repo - it's a bit heavyweight and the path may change...

Comment (Suraj2093) [reply wildfly-standalone-centos8/scripts/wildfly-setup.sh]: Removed Git clone and make required charges

Comment (Suraj2093) [reply wildfly-standalone-centos8/scripts/wildfly-setup.sh]: Made changes to script as exactly mentioned in the example to download the file, but I see the checks failing.

Comment (Suraj2093) [reply wildfly-standalone-centos8/scripts/wildfly-setup.sh]: Made a minor change in the script and now all the checks passed

**Comment by Suraj2093** [comment]
Can you please merge or please let me know if there is any action pending from my side.

**Comment by Suraj2093** [comment]
@bmoore-msft Can you please Merge this PR ?

**Pull request merged**